FirebaseFunctions/Backend/index.js

```diff
@@ -153,3 +157,29 @@ exports.genStreamError = functionsV2.https.onCall(
     }
   }
 );
+
+const weatherForecasts = {
+  Toronto: { conditions: 'snowy', temperature: 25 },
+  London: { conditions: 'rainy', temperature: 50 },
+  Dubai: { conditions: 'sunny', temperature: 75 }
+};
+
+async function* generateForecast(locations) {
+  for (const location of locations) {
+    yield { 'location': location,  ...weatherForecasts[location.name] };
+    await sleep(500);
+  }
+};
+
+exports.genStreamWeather = functionsV2.https.onCall(
+  async (request, response) => {
+    if (request.acceptsStreaming) {
+      for await (const chunk of generateForecast(request.data)) {
+        response.sendChunk({ chunk });
```

Comment:
@ncooke3 I think this is why you are seeing all SSE messages wrapped in "chunk: { JSON-data }".

If this way, instead, something like

```diff
- response.sendChunk({chunk})
+ response.sendChunk(chunk)
```

You won't see that wrapper

Reply:
Thanks, @taeold! This is important for us to understand since the client SDK needs to handle decoding all possibilities. Is this an accurate summary:

1. If `response.sendChunk(chunk)` is used, then `{ "message":  <JSON-data> }`
2. If `response.sendChunk({chunk})` is used, then `{ "message": { "chunk": <JSON-data> } }`
3. If something like `response.sendChunk({"chunk" : chunk})` is used, then `{ "message": { "chunk": { "chunk": <JSON-data> } } }`

Is it a similar case for the `result` or is that always: `{ "result": "The sky is blue because the sky is blue" }`

Reply:
Thanks, @taeold, makes sense. This actually simplifies the SDK's decoding logic a little bit. I thought the `chunk` was _always_ part of the response. It will be up to developers writing their CF3 to ensure that whatever they return can be decoded to the type their expecting within the SDK.